autofs-5.1.1 - fix left mount count return from umount_multi_triggers()

From: Xu Wang <xuw@redhat.com>

If a umount of an NFS mount at the root of a multi-mount fails
umount_multi_triggers() will return an zero (success) when it should
return 1 (a fail). In this case do_expire_direct() will close the
me->ioctlfd fd used for expires stopping further expires of the
direct mount.

Signed-off-by: Ian Kent <raven@themaw.net>
---
 CHANGELOG    |    1 +
 lib/mounts.c |    2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

--- autofs-5.0.5.orig/CHANGELOG
+++ autofs-5.0.5/CHANGELOG
@@ -254,6 +254,7 @@
 - fix fix map entry duplicate offset detection.
 - handle duplicates in multi mounts.
 - fix macro usage in lookup_program.c.
+- fix left mount count return from umount_multi_triggers().
 
 03/09/2009 autofs-5.0.5
 -----------------------
--- autofs-5.0.5.orig/lib/mounts.c
+++ autofs-5.0.5/lib/mounts.c
@@ -2225,7 +2225,7 @@ int umount_multi_triggers(struct autofs_
 				if (mount_multi_triggers(ap, me, root, strlen(root), "/") < 0)
 					warn(ap->logopt,
 					     "failed to remount offset triggers");
-				return left++;
+				return ++left;
 			}
 		}
 
